dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -34,26 +35,64 @@ const spaces = / /g;
 const ophanComponentId = (name: string) =>
 	name.toLowerCase().replace(spaces, '-');
 
+const getMerchHighPosition = (
+	collectionCount: number,
+	isNetworkFront: boolean | undefined,
+) => {
+	if (collectionCount < 4) {
+		return 2;
+	} else if (isNetworkFront) {
+		return 5;
+	} else {
+		return 4;
+	}
+};
+
 const shouldInsertMerchHigh = (
 	index: number,
-	isNetworkFront: boolean | undefined,
-	collectionCount: number,
 	isPaidContent: boolean | undefined,
+	collectionCount: number,
+	isNetworkFront: boolean | undefined,
 ) => {
 	const minContainers = isPaidContent ? 1 : 2;
 	if (collectionCount < minContainers) {
 		return false;
 	}
-	let desiredPosition;
-	if (collectionCount < 4) {
-		desiredPosition = 2;
-	} else if (isNetworkFront) {
-		desiredPosition = 5;
-	} else {
-		desiredPosition = 4;
-	}
 
-	return index === desiredPosition;
+	return index === getMerchHighPosition(collectionCount, isNetworkFront);
+};
+// On mobile, we remove the first container if it is a thrasher
+// and remove a container if it, or the next sibling, is a commercial container
+// we also exclude any containers that are directly before a thrasher
+// then we take every other container, up to a maximum of 10, for targeting MPU insertion
+
+const mobileAddPosition: number[] = [];
+
+const getMobileAddPosition = (
+	isNetworkFront: boolean | undefined,
+	collections: DCRCollectionType[],
+) => {
+	const merchHighPosition = getMerchHighPosition(
+		collections.length,
+		isNetworkFront,
+	);
+
+	collections.forEach((el, ind, arr) => {
+		if (
+			(ind === 0 && el.collectionType === 'fixed/thrasher') ||
+			ind === merchHighPosition ||
+			ind + 1 === merchHighPosition ||
+			arr[ind + 1]?.collectionType === 'fixed/thrasher'
+		) {
+			return false;
+		} else if (ind % 2 === 0) {
+			mobileAddPosition.push(ind);
+		}
+		return false;
+	});
+	if (mobileAddPosition.length > 10) {
+		mobileAddPosition.length = 10;
+	}
```

Comment:
I haven't tested this, but perhaps a better way to limit the length would be to slice the array. This would have to work in conjunction with my changes above, 


```ts
// Limit the number of ad positions to 10
mobileAdPosition = mobileAdPosition.slice(0, 10); 

```